### PR TITLE
feature(checkbox): Add ripple effect to checkbox

### PR DESCRIPTION
**Pull request by tinayuangao**
R: @jelbourn

### CODE REVIEW

File: src/lib/checkbox/checkbox.spec.ts [hunk 1]
```diff
@@ -483,6 +483,20 @@ describe('MdCheckbox', () => {
 
       expect(inputElement.tabIndex).toBe(13);
     });
+
+    it('should remove ripple if md-ripple-disabled input is set', async(() => {
+      testComponent.disableRipple = true;
+      fixture.detectChanges();
+
+      expect(checkboxNativeElement.querySelectorAll('[md-ripple]').length).toBe(0,
+        'Expect no [md-ripple] in checkbox');
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.spec.ts]: In general, prefer to break at the higher syntactic level, e.g.:
```
expect(checkboxNativeElement.querySelectorAll('[md-ripple]').length)
    .toBe(0, 'Expect no [md-ripple] in checkbox');
```

File: src/lib/checkbox/checkbox.ts [hunk 2]
```diff
@@ -333,10 +337,19 @@ export class MdCheckbox implements ControlValueAccessor {
...
     return `md-checkbox-anim-${animSuffix}`;
   }
+
+  getHostElement() {
+    return this._elementRef.nativeElement;
+  }
+
+  isRippleEnabled() {
+    return !this.disableRipple;
+  }
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.ts]: Why not bind to the `disableRipple` directly?

File: src/lib/checkbox/checkbox.ts [hunk 3]
```diff
@@ -88,6 +89,9 @@ export class MdCheckbox implements ControlValueAccessor {
   /** A unique id for the checkbox. If one is not supplied, it is auto-generated. */
   @Input() id: string = `md-checkbox-${++nextId}`;
 
+  /** Whether the ripple effect on click should be disabled. */
+  @Input() @BooleanFieldValue() disableRipple: boolean = false;
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.ts]: I just removed `@BooleanFieldValue` in another PR- custom decorators like this caused tree-shaking not to work. The replacement is a function called `coerceBooleanProperty`

**Comment by tinayuangao** [comment]
Comments addressed. Please take another look. Thanks!

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**